python/pacemaker/_cts/test.py

```diff
@@ -147,36 +162,38 @@ def _match_log_patterns(self):
             self._result_txt as appropriate.  Not all subclass will need to do
             this.
         """
-        if len(self._log_patterns) == 0 and len(self._negative_log_patterns) == 0:
+        if len(self._patterns) == 0:
             return
 
-        negative_matches = 0
-        cur = 0
-        pats = self._log_patterns
-        total_patterns = len(self._log_patterns)
+        n_failed_matches = 0
+        n_negative_matches = 0
 
-        for line in self._daemon_output.split("\n"):
-            negative_matches += self._count_negative_matches(line)
+        for pat in self._patterns:
+            positive_match = False
 
-            if len(pats) == 0:
-                continue
+            for line in self._daemon_output.split("\n"):
+                if pat.match(line) and pat.negative:
```

Comment:
We can improve the efficiency a little by doing the split before the outer loop, and checking pat.match() once per inner loop iteration

Reply:
Fixed.